modules/standard/Random.chpl

```diff
@@ -1172,14 +1172,14 @@ module Random {
                                    if ``prob`` has no non-zero values,
                                    if ``size < 1 || size.size < 1``,
                                    if ``replace=false`` and ``size > x.size || size.size > x.size``.
-                                   if ``isBoundedRange(x) == false``
+                                   if ``x`` does not have both bounds
      */
       proc choice(x: range(stridable=?), size:?sizeType=none, replace=true, prob:?probType=none)
         throws
       {
         var dom: domain(1,stridable=true);
 
-        if !isBoundedRange(x) {
+        if x.bounds != boundKind.both {
           compilerError('input range must be bounded');
```

Comment:
Probably out of scope for this PR, but this doesn't look like it throws an `IllegalArgumentError` as the chpldoc comment describes

Reply:
It looks to me as though the `_choice` helper that it calls does, so this could throw that onward?

Reply:
I think that's true for the other errors but not the error for an unbounded range, since it's halting with a `compilerError` right here

Reply:
Yes exactly, that case should be removed or the code modified to match the doc comment. By "this doesn't look like it throws" I meant specifically this case. Fixing it in this PR works for me.

Reply:
I ended up doing this last night in https://github.com/chapel-lang/chapel/pull/22079/commits/0c42ef8f0791ef32141f97077838a566e5a75f6a — thanks for catching it.